File: cases/taint/typescript/closure-capture-negative/ClosureCaptureNegative.ts
function dfb_source(): string { // DFB-SOURCE: closure-capture-input
  return "tainted";
}

function dfb_sink(value: string): void {} // DFB-SINK: closure-capture-sink

function makeHandler(): () => void {
  const tainted: string = dfb_source(); // DFB-WITNESS: closure-capture-store
  const captured: string = "clean";
  return (): void => {
    dfb_sink(captured);
  };
}

function run(): void {
  const handler: () => void = makeHandler();
  handler();
}

File: cases/taint/go/closure-capture-negative/closure_capture_negative.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: closure-capture-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: closure-capture-sink

func makeHandler() func() {
	tainted := dfb_source() // DFB-WITNESS: closure-capture-store
	_ = tainted
	captured := "clean"
	return func() {
		dfb_sink(captured)
	}
}

func run() {
	handler := makeHandler()
	handler()
}
